crypto/dsa/dsa_ameth.c

```diff
@@ -516,33 +516,16 @@ static int dsa_pkey_ctrl(EVP_PKEY *pkey, int op, long arg1, void *arg2)
 
 const EVP_PKEY_ASN1_METHOD dsa_asn1_meths[5] = {
 
-    {
-     EVP_PKEY_DSA2,
-     EVP_PKEY_DSA,
-     ASN1_PKEY_ALIAS},
+    {EVP_PKEY_DSA2, EVP_PKEY_DSA, ASN1_PKEY_ALIAS, "DSA-with-SHA"},
 
-    {
-     EVP_PKEY_DSA1,
-     EVP_PKEY_DSA,
-     ASN1_PKEY_ALIAS},
+    {EVP_PKEY_DSA1, EVP_PKEY_DSA, ASN1_PKEY_ALIAS, "DSA-2"},
 
-    {
-     EVP_PKEY_DSA4,
-     EVP_PKEY_DSA,
-     ASN1_PKEY_ALIAS},
+    {EVP_PKEY_DSA4, EVP_PKEY_DSA, ASN1_PKEY_ALIAS, "DSA-with-SHA1.2"},
 
-    {
-     EVP_PKEY_DSA3,
-     EVP_PKEY_DSA,
-     ASN1_PKEY_ALIAS},
+    {EVP_PKEY_DSA3, EVP_PKEY_DSA, ASN1_PKEY_ALIAS, "DSA-with-SHA1"},
```

Comment:
Where do these new names come from? I can't see a consistent naming pattern behind them and neither was I able to find these names in the source.

Reply:
They are my attempt at explaning the NID values in the constants.

Reply:
Ah, I see. Your naming scheme was inspired by the NID names:

```
include/openssl/evp.h:45:# define EVP_PKEY_DSA    NID_dsa
include/openssl/evp.h:46:# define EVP_PKEY_DSA1   NID_dsa_2
include/openssl/evp.h:47:# define EVP_PKEY_DSA2   NID_dsaWithSHA
include/openssl/evp.h:48:# define EVP_PKEY_DSA3   NID_dsaWithSHA1
include/openssl/evp.h:49:# define EVP_PKEY_DSA4   NID_dsaWithSHA1_2
```

But wouldn't it be better then to just use the literal NID names without the `NID_` prefix?

Alternatively, since the existing entry is a long name, `"OpenSSL DSA method"`, you could choose long names throughout:

```
{EVP_PKEY_DSA2, EVP_PKEY_DSA, ASN1_PKEY_ALIAS, "OpenSSL DSA method with SHA"},
{EVP_PKEY_DSA1, EVP_PKEY_DSA, ASN1_PKEY_ALIAS, "OpenSSL DSA method 2"},
{EVP_PKEY_DSA4, EVP_PKEY_DSA, ASN1_PKEY_ALIAS, "OpenSSL DSA method with SHA1/SHA2"},
{EVP_PKEY_DSA3, EVP_PKEY_DSA, ASN1_PKEY_ALIAS, "OpenSSL DSA method with SHA1"},
{EVP_PKEY_DSA,  EVP_PKEY_DSA, 0, "DSA", "OpenSSL DSA method",
```

(BTW: does SHA1/SHA2 mean SHA1 _or_ SHA2?)

Reply:
Or does `SHA1_2` stand for `OpenSSL DSA method with SHA1 (second attempt)`? Sigh! ASN1 is a mess...